lib/chef/resource/apt_repository.rb

```diff
@@ -422,7 +422,7 @@ def build_repo(uri, distribution, components, trusted, arch, options, add_src =
                      "[" + options_list.join(" ") + "]"
                    end
           info = [ optstr, uri.normalize.to_s, distribution, components ].compact.join(" ")
-          repo =  "deb      #{info}\n"
+          repo =  "deb #{info}\n"
```

Comment:
If I understand correctly (and please let me know if I've misunderstood here), this will drop spaces for people who already ran it like this... that seems wrong, it'll cause diffs for everyone else.

More specifically - this seems like another process is re-writing your repo file with different spacing - do you have two different processes messing with that file? That's definitely not a supported configuration.

Reply:
Thanks for looking at this. I only have chef writing to these files, and the files are written properly, without any extra spaces. Here's what I have in one of those files right now:

`deb http://apt.postgresql.org/pub/repos/apt jammy-pgdg main`

Which is as you would expect.

But every time I run chef-client, I get a diff, as if the file had changed.

Reply:
I'll add that the spaces I see are not introduced in the recipe either. This is the apt_repository block:

```
apt_repository 'postgresql' do
  uri 'http://apt.postgresql.org/pub/repos/apt'
  distribution 'jammy-pgdg'
  components ['main']
  key 'https://www.postgresql.org/media/keys/ACCC4CF8.asc'
  action :add
end
```

Reply:
Upon further discussion, this would seem to be best handled via enabling a way to produce an "ignore whitespace" diff from `Chef::Util::Diff`. Unfortunately, that would likely involve a bit of replumbing the diff in the file resource, and the whitespace appears to be an intentional formatting choice:

```ruby
          repo =  "deb      #{info}\n"
          repo << "deb-src  #{info}\n" if add_src
```

Changing this will likely result in everyone's `sources.lst` getting updated upon the next run. Are there deb sources/packages that are in addition to the defaults that might be consuming and reformatting the sources.lst upon install/update?

Reply:
Should this be closed as won't implement?

Reply:
I just did several tests and added in lots of spaces and cannot repro this. It seems very unlikely that the file resource generates the file in different ways, but I'll assign this to myself and see if I can repro this in anyway.

Reply:
Using your exactly-specific resource, I cannot reproduce this. It is very certain you have something else generating that file, perhaps even a similar resource in the same chef run.

```
root@rider:~# cat /tmp/test.rb 
apt_repository 'postgresql' do
  uri 'http://apt.postgresql.org/pub/repos/apt'
  distribution 'jammy-pgdg'
  components ['main']
  key 'https://www.postgresql.org/media/keys/ACCC4CF8.asc'
  action :add
end
root@rider:~# chef-apply /tmp/test.rb 
Recipe: (chef-apply cookbook)::(chef-apply recipe)
  * apt_repository[postgresql] action add
    * execute[apt-cache gencaches] action nothing (skipped due to action :nothing)
    * apt_update[postgresql] action nothing (skipped due to action :nothing)
    * remote_file[/var/chef/cache/https___www_postgresql_org_media_keys_ACCC4CF8_asc] action create (up to date)
    * execute[apt-key add /var/chef/cache/https___www_postgresql_org_media_keys_ACCC4CF8_asc] action run (skipped due to not_if)
    * file[/etc/apt/sources.list.d/postgresql.list] action create (up to date)
     (up to date)
root@rider:~# chef-apply /tmp/test.rb 
Recipe: (chef-apply cookbook)::(chef-apply recipe)
  * apt_repository[postgresql] action add
    * execute[apt-cache gencaches] action nothing (skipped due to action :nothing)
    * apt_update[postgresql] action nothing (skipped due to action :nothing)
    * remote_file[/var/chef/cache/https___www_postgresql_org_media_keys_ACCC4CF8_asc] action create (up to date)
    * execute[apt-key add /var/chef/cache/https___www_postgresql_org_media_keys_ACCC4CF8_asc] action run (skipped due to not_if)
    * file[/etc/apt/sources.list.d/postgresql.list] action create (up to date)
     (up to date)
```